A66-otel-stats.md

```diff
@@ -0,0 +1,540 @@
+    [views](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view) through
+    the MeterProvider.
+*   Optionally allow setting of a OpenTelemetry plugin for a specific channel or
+    server, instead of setting it globally.
+*   Optionally allowing setting of a map of constant attributes that are
+    recorded on all metrics associated with that plugin.
+
+Note that implementations of the gRPC OpenTelemetry plugin
+[should prefer](https://opentelemetry.io/docs/specs/otel/overview/) to only
+depend on the OpenTelemetry API and not the OpenTelemetry SDK.
+
+The [Meter](https://opentelemetry.io/docs/specs/otel/metrics/api/#get-a-meter)
+creation should use a `name` that identifies the library, for example,
+"grpc-c++", "grpc-java", "grpc-go". The `version` should be the same as the
+release version of the gRPC library, for example, "1.57.1".
+
+#### C++
+
+```c++
+class OpenTelemetryPluginBuilder {
+ public:
+  OpenTelemetryPluginBuilder();
+  // If `SetMeterProvider()` is not called, no metrics are collected.
+  OpenTelemetryPluginBuilder& SetMeterProvider(
+      std::shared_ptr<opentelemetry::metrics::MeterProvider> meter_provider);
+  // Methods to manipulate which instruments are enabled in the OTel Stats
+  // Plugin. The default set of instruments are -
+  // grpc.client.attempt.started
+  // grpc.client.attempt.duration
+  // grpc.client.attempt.sent_total_compressed_message_size
+  // grpc.client.attempt.rcvd_total_compressed_message_size
+  // grpc.server.call.started
+  // grpc.server.call.duration
+  // grpc.server.call.sent_total_compressed_message_size
+  // grpc.server.call.rcvd_total_compressed_message_size
+  OpenTelemetryPluginBuilder& EnableMetric(absl::string_view metric_name);
+  OpenTelemetryPluginBuilder& DisableMetric(absl::string_view metric_name);
+  OpenTelemetryPluginBuilder& DisableAllMetrics();
+  // If set, \a target_attribute_filter is called per channel to decide whether
+  // to record the target attribute on client or to replace it with "other".
+  // This helps reduce the cardinality on metrics in cases where many channels
+  // are created with different targets in the same binary (which might happen
+  // for example, if the channel target string uses IP addresses directly).
+  OpenTelemetryPluginBuilder& SetTargetAttributeFilter(
+      absl::AnyInvocable<bool(absl::string_view /*target*/) const>
+          target_attribute_filter);
+  // If set, \a generic_method_attribute_filter is called per call with a
+  // generic method type to decide whether to record the method name or to
+  // replace it with "other". Non-generic or pre-registered methods remain
+  // unaffected. If not set, by default, generic method names are replaced with
+  // "other" when recording metrics.
+  OpenTelemetryPluginBuilder& SetGenericMethodAttributeFilter(
+      absl::AnyInvocable<bool(absl::string_view /*generic_method*/) const>
+          generic_method_attribute_filter);
+  void BuildAndRegisterGlobal();
```

Comment:
Can we add comments to explain what this API does?

Reply:
Done